src/nvim/lua/xdiff.c

```diff
@@ -205,6 +262,12 @@ static NluaXdiffMode process_xdl_diff_opts(lua_State *lstate, xdemitconf_t *cfg,
         goto exit_1;
       }
       cfg->interhunkctxlen = v->data.integer;
+    } else if (strequal("linematch", k.data)) {
+      if (v->data.boolean) {
```

Comment:
use `api_object_to_bool()` (or if we intent on lua style thruthiness here, need to check `v->type` first, accessing non-current union member is UB)